src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -495,14 +551,23 @@ private void publishSurveyLaunchedEvent(UUID caseId, String questionnaireId, Str
         .debug("SurveyLaunch event published");
   }
 
-  private void publishNewAddressReportedEvent(UUID caseId, AddressIndexAddressCompositeDTO address)
+  private void publishNewAddressReportedEvent(
```

Comment:
I know much of this method is not your original code, but there are a few observations here:
- no code coverage for the IllegalArgumentException branch
- The IllegalArgument code is essentially a cut-n-paste version of very similar code further below. It would be good to combine.
- on line 597, should we be setting the NewAddress sourceCaseId element ?

Reply:
Not at all keen on creating extracting into a method as don't think that the benefits outweight the disadvantages (only 2 statements used twice).
The exception handling is interesting though. Looks like there is no test coverage of the exception branch because it can't actually be executed!
Moving the check/through into address service code. Better to validate once as we receive data than every part of the code doing it on an ad-hoc basis.

Reply:
Can't set the sourceCaseId as we don't have one. Wasn't being set before either.

Reply:
We don't need to set sourceCaseId